File: trunk/JAVA/PokerProtocol/src/bluffinmuffin/protocol/commands2/lobby/training/IdentifyCommand.java
package bluffinmuffin.protocol.commands2.lobby.training;

import java.util.StringTokenizer;

import bluffinmuffin.protocol.commands2.lobby.AbstractLobbyCommand;

public class IdentifyCommand extends AbstractLobbyCommand
{
    @Override
    protected String getCommandName()
    {
        return IdentifyCommand.COMMAND_NAME;
    }
    
    public static String COMMAND_NAME = "lobbyTRAINING_IDENTIFY";
    
    private final String m_playerName;
    
    public IdentifyCommand(StringTokenizer argsToken)
    {
        m_playerName = argsToken.nextToken();
    }
    
    public IdentifyCommand(String name)
    {
        m_playerName = name;
    }
    
    @Override
    public void encode(StringBuilder sb)
    {
        append(sb, m_playerName);
    }
    
    public String encodeResponse(Boolean success)
    {
        return new IdentifyResponse(this, success).encode();
    }
    
    public String getPlayerName()
    {
        return m_playerName;
    }
}

File: trunk/C#/PokerProtocol/Commands/Lobby/Training/IdentifyCommand.cs
using System;
using System.Collections.Generic;
using EricUtility;
using EricUtility.Networking.Commands;
using System.Text;
using Newtonsoft.Json.Linq;

namespace PokerProtocol.Commands.Lobby.Training
{
    public class IdentifyCommand : AbstractLobbyCommand
    {
        public static string COMMAND_NAME = "lobbyTRAINING_IDENTIFY";

        public string Name { get; set; }

        public IdentifyCommand()
        {
        }

        public IdentifyCommand(string name)
        {
            Name = name;
        }

        public string EncodeResponse( bool success )
        {
            return new IdentifyResponse(this, success).Encode();
        }
    }
}
